dm/syncer/safe_mode.go

```diff
@@ -43,35 +57,77 @@ func (s *Syncer) enableSafeModeInitializationPhase(tctx *tcontext.Context) {
 		s.safeMode.Add(tctx, 1) // add 1 but has no corresponding -1, so keeps enabled
 		s.tctx.L().Info("enable safe-mode by config")
 	}
-	if s.checkpoint.SafeModeExitPoint() != nil {
+
+	var duration time.Duration
+	initPhaseSeconds := s.cfg.SafeModeDuration
+
+	failpoint.Inject("SafeModeInitPhaseSeconds", func(val failpoint.Value) {
+		initPhaseSeconds = val.(string)
+		s.tctx.L().Info("set initPhaseSeconds", zap.String("failpoint", "SafeModeInitPhaseSeconds"), zap.String("value", initPhaseSeconds))
+	})
+	failpoint.Inject("SafeModeDurationSet", func(val failpoint.Value) {
+		initPhaseSeconds = val.(string)
+		s.tctx.L().Info("set initPhaseSeconds", zap.String("failpoint", "SafeModeDurationSet"), zap.String("value", initPhaseSeconds))
+	})
```

Comment:
This two failpoint do the same thing

Reply:
`SafeModeInitPhaseSeconds` was used in so many places. So `SafeModeInitPhaseSeconds` keep the previous logic.

`SafeModeDurationSet` only change the safe-mode-duration